src/components/Errors/Errors.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useRef } from 'react';
+import { useTodos } from '../../utils/TodoContext';
+import classNames from 'classnames';
+
+export const Errors: React.FC = () => {
+  const { error, clearCompletedError, setError, setClearCompletedError } =
+    useTodos();
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+  const combinedError = error || clearCompletedError;
+
+  useEffect(() => {
+    if (combinedError) {
+      if (timerRef.current) {
+        clearTimeout(timerRef.current);
+      }
+
+      timerRef.current = setTimeout(() => {
+        setError(null);
+        setClearCompletedError(null);
+        timerRef.current = null;
+      }, 3000);
+    }
+
+    return () => {
+      if (timerRef.current) {
+        clearTimeout(timerRef.current);
+        timerRef.current = null;
+      }
```

Comment:
You don't need to have ref to keep timer id. You can simply define variable using `let` inside body of `effect`